File: tests/test-terraria/src/main/java/pize/tests/terraria/graphics/MapRenderInfo.java
package pize.tests.terraria.graphics;

import pize.Pize;

public class MapRenderInfo{

    public static final int TILES_PER_SCREEN = 30;

    private float cellSize, cellsPerWidth, cellsPerHeight, scale;

    public MapRenderInfo(){
        scale = 1;
    }


    public void update(){
        int windowWidth = Pize.getWidth();
        int windowHeight = Pize.getHeight();

        if(windowWidth > windowHeight)
            cellSize = (float) windowWidth / TILES_PER_SCREEN;
        else
            cellSize = (float) windowHeight / TILES_PER_SCREEN;

        cellsPerWidth = windowWidth / cellSize;
        cellsPerHeight = windowHeight / cellSize;
    }


    public float getCellSize(){
        return cellSize;
    }

    public float getCellsPerWidth(){
        return cellsPerWidth;
    }

    public float getCellsPerHeight(){
        return cellsPerHeight;
    }


    public float getScale(){
        return scale;
    }

    public void mulScale(double scale){
        this.scale *= scale;
    }

    public void setScale(float scale){
        this.scale = scale;
    }

}

File: tests/test-terraria/src/main/kotlin/pize/tests/terraria/graphics/MapRenderInfo.kt
package pize.tests.terraria.graphics

import pize.Pize.height
import pize.Pize.width

class MapRenderInfo {
    var cellSize = 0f
        private set
    var cellsPerWidth = 0f
        private set
    var cellsPerHeight = 0f
        private set
    var scale = 1f

    fun update() {
        val windowWidth = width
        val windowHeight = height
        cellSize =
            if (windowWidth > windowHeight) windowWidth.toFloat() / TILES_PER_SCREEN else windowHeight.toFloat() / TILES_PER_SCREEN
        cellsPerWidth = windowWidth / cellSize
        cellsPerHeight = windowHeight / cellSize
    }

    fun mulScale(scale: Double) {
        this.scale *= scale.toFloat()
    }

    companion object {
        const val TILES_PER_SCREEN = 30
    }
}
